app/jobs/clean_up_expired_pushes_after_duration_job.rb

```diff
@@ -0,0 +1,63 @@
+class CleanUpExpiredPushesAfterDurationJob < ApplicationJob
+  queue_as :default
+
+  # Delete Expired Pushes After Duration
+  #
+  # When a push expires, the payload is deleted but the metadata record still exists.  This
+  # includes information such as creation date, audit logs, duration etc..  If purge_after setting
+  # is set to a duration, this job will delete expired pushes and their audit logs after
+  # that duration.
+  #
+  # If a user attempts to retrieve a secret link that doesn't exist anymore, we still show
+  # the standard "This secret link has expired" message.  This strategy provides two benefits:
+  #
+  # 1. It hides the fact that if a secret ever exists or not (more secure)
+  # 2. It allows us to delete data that we don't want
+  #
+  # This task will run through all records expired more than purge_after ago and delete them entirely.
+  #
+  # Because of the above, expired secret URLs still will show the same
+  # expiration message.
+  #
+  def perform(*args)
+    return if Settings.purge_after == "disabled" || to_duration(Settings.purge_after).nil?
+
+    # Log task start
+    logger.info("--> #{self.class.name}: Starting...")
+
+    counter = 0
+
+    Push.includes(:audit_logs)
+      .where(expired: true)
+      .where("expired_on < ?", Time.current - to_duration(Settings.purge_after))
+      .find_each do |push|
+      counter += 1
+      push.destroy
+    end
+
+    logger.info("  -> #{counter} total pushes expired more than #{Settings.purge_after} ago were deleted.")
+
+    # Log completion
+    logger.info("  -> #{self.class.name}: Finished.")
+  end
+
+  private
+
+  # Convert a string duration to a time duration
+  def to_duration(str)
+    str.to_s.strip.split(" ").then { |quantity, unit| quantity.to_i.send(unit.downcase.to_sym) }
+  rescue
+    logger.error("Invalid duration: #{str}")
+
+    nil
```

Comment:
I haven't raised an error when the server is initialized. Do you want me to add an initializer to check the value of the purge_after setting @pglombardo .

Reply:
I agree/good idea.

Reply:
An initializer is added to check the `purge_after` setting